### PR TITLE
Make adjustable top/bottom margin

### PR DESCRIPTION
**Pull request by uivlis**
The margin delimits both the header/footer and the rest of the report. The rest of the report (text, ...) is positioned according to the header/footer size, so that they don't overlap. The improvement comes for all types of header/footer, except for ReportFooterOnPages.ALL_BUT_LAST

### CODE REVIEW

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java [hunk 1]
```diff
@@ -16,7 +17,10 @@
     private float marginTop;
     private float marginBottom;
     private float lineDistance;
+    private float header;
     private float footer;
+    private ReportFooterOnPages footerOnPages;
+    private ReportFooterOnPages headerOnPages;
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java]: that sounds weird. a class ReportFooterOnPages with the variable name headerOnPages? what is the intention of that?

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/impl/PdfReportBuilderImpl.java [hunk 2]
```diff
@@ -134,6 +134,13 @@ public PdfReportStructure buildReport(PdfPageLayout pageConfig) {
         }
         for (AbstractFixedLineGenerator generator : fixedLineGenerators) {
             generator.addFooterToAllPages(report, pageConfig);
+            if (generator instanceof PdfFooterGenerator) {
+                pageConfig.setFooterOnPages(generator.getFooterOnPages());
+                pageConfig.setFooter(generator.getFooterElement().getHeight(pageConfig.getUsableWidth()) + configuration.getLineDistance());
+            }else {
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/impl/PdfReportBuilderImpl.java]: code format

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java (hunk 1 above)
Comment (uivlis) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java]: Well, at a certain point in the code there is no distinction between header and footer, and so ReportFooterOnPages applies to both

Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java]: I dont like those variable names, it is confusing, I dont know what it should be

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/utils/ReportStaticElementOnPages.java [hunk 3]
```diff
@@ -3,6 +3,6 @@
 /**
  * Created by sfarcas on 7/21/2016.
  */
-public enum ReportFooterOnPages {
-    ALL, ALL_BUT_FIRST, ALL_BUT_LAST;
+public enum ReportStaticElementOnPages {
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/utils/ReportStaticElementOnPages.java]: sometimes javadoc could really help. explain the meaning of that enum here.

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/utils/ReportStaticElementOnPages.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package cc.catalysts.boot.report.pdf.utils;
+
+/**
+ * Created by sfarcas on 7/21/2016.
+ *
+ * position of the footer or the header:
+ * on all pages, on all pages except the first, on all pages except the last
+ *
+ */
+public enum ReportStaticElementOnPages {
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/utils/ReportStaticElementOnPages.java]: why not:

enum PositionOfStaticElements {
    OnAllPages,
    OnAllPagesButFirst,
    OnAllPagesButLast,
}

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java [hunk 5]
```diff
@@ -28,17 +27,17 @@
 
     PdfReportBuilder withHeaderOnAllPages(ReportElement headerElement);
 
-    PdfReportBuilder withHeaderOnPages(String left, String middle, String right, ReportFooterOnPages headerOnPages);
+    PdfReportBuilder withHeaderOnPages(String left, String middle, String right, PositionOfStaticElements headerOnPages);
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/PdfReportBuilder.java]: headerOnPages --> positionOfHeader

File: cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java [hunk 6]
```diff
@@ -16,7 +17,10 @@
     private float marginTop;
     private float marginBottom;
     private float lineDistance;
+    private float header;
     private float footer;
+    private PositionOfStaticElements footerOnPages;
```
Comment (klu2) [inline cat-boot-report-pdf/src/main/java/cc/catalysts/boot/report/pdf/config/PdfPageLayout.java]: positionOfFooter

**Pull request merged**